dotnet/src/IntegrationTests/Connectors/OpenAI/OpenAIChatCompletion_NoneFunctionChoiceBehaviorTests.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+using SemanticKernel.IntegrationTests.TestSettings;
+using Xunit;
+
+namespace SemanticKernel.IntegrationTests.Connectors.OpenAI;
+
+public sealed class OpenAINoneFunctionChoiceBehaviorTests : BaseIntegrationTest
+{
+    private readonly Kernel _kernel;
+    private readonly FakeFunctionFilter _autoFunctionInvocationFilter;
+
+    public OpenAINoneFunctionChoiceBehaviorTests()
+    {
+        this._autoFunctionInvocationFilter = new FakeFunctionFilter();
+
+        this._kernel = this.InitializeKernel();
+        this._kernel.AutoFunctionInvocationFilters.Add(this._autoFunctionInvocationFilter);
+    }
+
+    [Fact]
+    public async Task SpecifiedInCodeInstructsConnectorNotToInvokeKernelFunctionAsync()
+    {
+        // Arrange
+        var plugin = this._kernel.CreatePluginFromType<DateTimeUtils>();
+        this._kernel.Plugins.Add(plugin);
+
+        var invokedFunctions = new List<string>();
+
+        this._autoFunctionInvocationFilter.RegisterFunctionInvocationHandler(async (context, next) =>
+        {
+            invokedFunctions.Add(context.Function.Name);
+            await next(context);
+        });
+
+        // Act
+        var settings = new OpenAIPromptExecutionSettings() { FunctionChoiceBehavior = FunctionChoiceBehavior.None() };
+
+        var result = await this._kernel.InvokePromptAsync("How many days until Christmas?", new(settings));
+
+        // Assert
+        Assert.NotNull(result);
+
+        Assert.Empty(invokedFunctions);
+    }
+
+    //[Fact]
+    //public async Task SpecifiedInPromptInstructsConnectorNotToInvokeKernelFunctionAsync()
+    //{
+    //    // Arrange
+    //    this._kernel.ImportPluginFromType<DateTimeUtils>();
+
+    //    var invokedFunctions = new List<string>();
+
+    //    this._autoFunctionInvocationFilter.RegisterFunctionInvocationHandler(async (context, next) =>
+    //    {
+    //        invokedFunctions.Add(context.Function.Name);
+    //        await next(context);
+    //    });
+
+    //    var promptTemplate = """"
+    //        template_format: semantic-kernel
+    //        template: How many days until Christmas?
+    //        execution_settings:
+    //          default:
+    //            temperature: 0.1
+    //            function_choice_behavior:
+    //              type: none
+    //        """";
+
+    //    var promptFunction = KernelFunctionYaml.FromPromptYaml(promptTemplate);
+
+    //    // Act
+    //    var result = await this._kernel.InvokeAsync(promptFunction);
+
+    //    // Assert
+    //    Assert.NotNull(result);
+
+    //    Assert.Empty(invokedFunctions);
+    //}
```

Comment:
Will this be uncommented in a follow up PR?

Reply:
Yes, this test and a few others that are commented out will be uncommented in one of the following PRs.